src/test/java/uk/ac/ebi/ena/webin/cli/manifest/processor/metadata/SampleProcessorTest.java

```diff
@@ -12,43 +12,218 @@
-SampleProcessorTest
-{
+SampleProcessorTest {
     private final WebinCliParameters parameters = WebinCliTestUtils.getTestWebinCliParameters();
+    private final Logger log = LoggerFactory.getLogger(SampleProcessorTest.class);
+    private static final boolean TEST = true;
+
+    @Test
+    public void
+    testCorrect() {
+        SampleProcessor processor = new SampleProcessor(parameters,
+                (Sample sample) -> Assert.assertEquals("SAMEA749881", sample.getBioSampleId()));
 
-    @Test public void 
-    testCorrect()
-    {
-        SampleProcessor processor = new SampleProcessor( parameters,
-            (Sample sample) -> Assert.assertEquals( "SAMEA749881", sample.getBioSampleId() ) );
+        ManifestFieldValue fieldValue = createFieldValue(ManifestFieldType.META, "SAMPLE", "ERS000002");
+        ValidationResult result = new ValidationResult();
+        processor.process(result, fieldValue);
+        Assert.assertTrue(result.isValid());
+        Assert.assertEquals("SAMEA749881", fieldValue.getValue());
+    }
+
+
+    @Test
+    public void
+    testIncorrect() {
+        SampleProcessor processor = new SampleProcessor(parameters, Assert::assertNull);
+        ManifestFieldValue fieldValue = createFieldValue(ManifestFieldType.META, "SAMPLE", "SRP000392");
+        ValidationResult result = new ValidationResult();
+        processor.process(result, fieldValue);
+        Assert.assertFalse(result.isValid());
+        Assert.assertEquals("SRP000392", fieldValue.getValue());
+    }
+
+
+    /**
+     * Test Add and Modify with setSampleUpdate parameter.
+     * The collectionDate in sample object will be updated to current date. 
+     **/
+    @Test
+    public void
+    testJsonSampleAddAndUpdate() throws JsonProcessingException {
         
-        ManifestFieldValue fieldValue = createFieldValue( ManifestFieldType.META, "SAMPLE", "ERS000002" );
+        String sampleTestAlias = UUID.randomUUID().toString();
+       
+        // Create a new sample
+        SampleProcessor processor = new SampleProcessor(parameters,
+                (Sample sample) -> {
+                    Assert.assertNotNull(sample.getBioSampleId());
+                    Assert.assertEquals(getSampleCollectionDate(sample), "2010-01-20");
+                });
+        ManifestFieldValue fieldValue = createFieldValue(ManifestFieldType.META, "SAMPLE", getSampleJson(sampleTestAlias));
         ValidationResult result = new ValidationResult();
-        processor.process( result, fieldValue );
-        Assert.assertTrue(result.isValid() );
-        Assert.assertEquals( "SAMEA749881", fieldValue.getValue() );
+        processor.process(result, fieldValue);
+        Assert.assertTrue(result.isValid());
+        Assert.assertNotNull(fieldValue.getValue().startsWith("SAMEA"));
+        String biosampleId = fieldValue.getValue();
+        
+        // Update sample
+        JsonNode sampleJson = new ObjectMapper().readTree(getSampleJson(sampleTestAlias));
```

Comment:
We could use the sample object here from webin-xml.

String modifiedCollectionDate =  LocalDate.now().toString();